src/test/java/de/tum/in/www1/artemis/TestRepositoryResourceIntegrationTest.java

```diff
@@ -0,0 +1,289 @@
+        request.postWithoutLocation(testRepoBaseUrl + programmingExercise.getId() + "/commit", null, HttpStatus.OK, null);
+
+        // Check status of git after the commit
+        var receivedStatusAfterCommit = request.get(testRepoBaseUrl + programmingExercise.getId(), HttpStatus.OK, RepositoryStatusDTO.class);
+        assertThat(receivedStatusAfterCommit.repositoryStatus.toString()).isEqualTo("CLEAN");
+
+        // Create file in the local repository and commit it
+        Path localFilePath = Paths.get(testRepo.localRepoFile + "/" + fileName);
+        var localFile = Files.createFile(localFilePath).toFile();
+        // write content to the created file
+        FileUtils.write(localFile, "local");
+        gitService.stageAllChanges(localRepo);
+        testRepo.localGit.commit().setMessage("local").call();
+
+        // Create file in the remote repository and commit it
+        Path remoteFilePath = Paths.get(testRepo.originRepoFile + "/" + fileName);
+        var remoteFile = Files.createFile(remoteFilePath).toFile();
+        // write content to the created file
+        FileUtils.write(remoteFile, "remote");
+        gitService.stageAllChanges(remoteRepo);
+        testRepo.originGit.commit().setMessage("remote").call();
+
+        // Merge the two and a conflict will occur
+        testRepo.localGit.fetch().setRemote("origin").call();
+        List<Ref> refs = testRepo.localGit.branchList().setListMode(ListBranchCommand.ListMode.REMOTE).call();
+        var result = testRepo.localGit.merge().include(refs.get(0).getObjectId()).setStrategy(MergeStrategy.RESOLVE).call();
+        var status = testRepo.localGit.status().call();
+        assertThat(status.getConflicting().size() > 0).isTrue();
+        assertThat(MergeResult.MergeStatus.CONFLICTING).isEqualTo(result.getMergeStatus());
+
+        // Execute the reset Rest call
+        request.postWithoutLocation(testRepoBaseUrl + programmingExercise.getId() + "/reset", null, HttpStatus.OK, null);
+
+        // Check the git status after the reset
+        status = testRepo.localGit.status().call();
+        assertThat(status.getConflicting().size() == 0).isTrue();
+        assertThat(LocalRepository.getAllCommits(testRepo.localGit).get(0)).isEqualTo(LocalRepository.getAllCommits(testRepo.originGit).get(0));
+        var receivedStatusAfterReset = request.get(testRepoBaseUrl + programmingExercise.getId(), HttpStatus.OK, RepositoryStatusDTO.class);
+        assertThat(receivedStatusAfterReset.repositoryStatus.toString()).isEqualTo("CLEAN");
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void shouldGetStatus() throws Exception {
+        programmingExerciseRepository.save(programmingExercise);
+        var receivedStatus = request.get(testRepoBaseUrl + programmingExercise.getId(), HttpStatus.OK, RepositoryStatusDTO.class);
+        assertThat(receivedStatus).isNotNull();
```

Comment:
can you be a bit more specific here when testing for the correct status?

Reply:
i adapted the testcase